libs/@hashintel/design-system/src/block-prompt-input.tsx

```diff
@@ -0,0 +1,223 @@
+import {
+  Box,
+  Button,
+  buttonBaseClasses,
+  Collapse,
+  inputBaseClasses,
+  outlinedInputClasses,
+  SxProps,
+  TextField,
+  TextFieldProps,
+  Theme,
+} from "@mui/material";
+import {
+  FormEvent,
+  forwardRef,
+  FunctionComponent,
+  ReactNode,
+  useRef,
+  useState,
+} from "react";
+
+import { BlockErrorMessage } from "./block-error-message";
+
+type BlockPromptInputProps = {
+  error?: boolean;
+  apiName?: string;
+  buttonLabel?: ReactNode;
+  buttonSx?: SxProps<Theme>;
+  onSubmit?: () => void;
+} & TextFieldProps;
+
+export const BlockPromptInput: FunctionComponent<BlockPromptInputProps> =
+  forwardRef(
+    (
+      {
+        error,
+        apiName,
+        buttonLabel,
+        sx,
+        buttonSx,
+        onSubmit,
+        onChange,
+        disabled,
+        ...props
+      },
+      ref,
+    ) => {
+      const inputRef = useRef<HTMLTextAreaElement | null>();
+      const [hasMultipleLines, setHasMultipleLines] = useState(false);
+
+      const calculateMultipleLines = () => {
+        if (inputRef.current) {
+          const computedStyles = window.getComputedStyle(
+            inputRef.current,
+            null,
+          );
+
+          const lineHeight = parseInt(
+            computedStyles.getPropertyValue("line-height"),
+            10,
+          );
+          const paddingLeft = parseInt(
+            computedStyles.getPropertyValue("padding-left"),
+            10,
+          );
+          const paddingRight = parseInt(
+            computedStyles.getPropertyValue("padding-right"),
+            10,
+          );
+
+          const width = inputRef.current.offsetWidth;
+
+          // Calculate width and height of the text inside the input
+          const temp = document.createElement("span");
+          temp.style.fontSize = computedStyles.fontSize;
+          temp.style.fontFamily = computedStyles.fontFamily;
+          temp.style.lineHeight = computedStyles.lineHeight;
+          temp.style.maxWidth = computedStyles.width;
+          temp.style.wordBreak = "break-word";
+          temp.style.whiteSpace = "break-spaces";
+          temp.textContent = inputRef.current.value;
+          document.body.appendChild(temp);
+          const textWidth = temp.offsetWidth;
+          const textHeight = temp.offsetHeight;
+          document.body.removeChild(temp);
+
+          const firstLineOverflowsButton =
+            width -
+              paddingLeft -
+              paddingRight -
+              textWidth -
+              (hasMultipleLines ? 170 : 0) <
+            0;
+
+          const multipleLines = textHeight / lineHeight > 1;
+
+          setHasMultipleLines(firstLineOverflowsButton || multipleLines);
+        }
+      };
+
+      const submit = (event: FormEvent) => {
```

Comment:
Nit: we should make use of `useCallback` here

Reply:
Done in 80716f52603933117289cc7c14c682bd155f49f5